### DISCUSSION TITLE
segment anything support

### DISCUSSION THREAD

**Discussion by originlake** [wkentaro/labelme#1260] [Show and tell]
I spent some time trying to integrate [segment anything](https://github.com/facebookresearch/segment-anything) into labelme. And here is what I got, a few clicks to get good enough polygons.
![segment-anything-demo](https://user-images.githubusercontent.com/11423220/231196323-b8818b64-bf12-45a3-b254-a1db05017d07.gif)
It's my first time working with Qt and I did plenty of hacks in the code to make things work. Those hacks would probably break the original design so I don't plan to create a PR, but I'd like to share this interesting work.

**Comment by vietanhdev** [comment]
Please check out [LabelAnything](https://github.com/vietanhdev/anylabeling) - A fork of labelme with Segment Anything and YOLO models for auto labeling. 

Training loop instructions will be added soon.
![image](https://user-images.githubusercontent.com/18329471/232065045-683a61de-e363-4343-8946-addcdd529eb6.png)

Demo video: https://www.youtube.com/watch?v=5iQSGL7ebXE

**Comment by healthonrails** [comment]
Your incredible work is much appreciated! I've integrated it into [annolid](https://github.com/healthonrails/annolid) and created a demo showcasing its capabilities in action: [segement-anything](https://github.com/healthonrails/annolid/blob/main/docs/imgs/annolid_with_segment_anything.gif).

**Comment by wkentaro** [comment]
I'm working on the integration of SAM with Labelme in here: https://github.com/wkentaro/labelme/pull/1262.
It is still an alpha version, but that would be great if you can try and give feedback. (the PR has installation/usage guides)
